robot-valjak (#2)
pfe-rs/izvestaji: content/2022/robot-valjak.md

title: Robot valjak
summary: Robot u obliku valjka je projekat rađen na letnjem kampu za stare polaznike 2022. godine od Nikole Ristanovića i Filipa Bajraktarija.
---

> 

Reviewer: Dodajte apstrakt

Druga dva merenja predstavljaju kako u realnom sistemu napon na motoru utiče na ugaonu brzinu klatna i ugao inklinacije klatna. Na drugom grafiku se može jasno videti gore opisan problem. Referentna nula je pomerena za neku vrednost $d\theta$ i razlikuje se od stvarne nule. Takođe, jedan od ciljeva ovog projekta je bila i numerička identifikacija sistema koja nažalost nije urađena zbog nedostatka vremena kao i gore navedenih problema, ali idalje ostaje kao mogućnost za neki budući rad na ovu ili neku sličnu temu.

> ## Zaključak

Reviewer: Dodajte zaključak

Author: Dodat je zaključak.

title: Robot valjak
summary: Robot u obliku valjka je projekat rađen na letnjem kampu za stare polaznike 2022. godine od Nikole Ristanovića i Filipa Bajraktarija.
---

> ![Grafički apstrakt](/images/2022/robot-valjak/graficki_apstrakt.svg)

Reviewer: * Zamenite imena signala sa simbolima.
* Koristite sans serif i veći font (da se lakše čita)

| $\ \ \ \ \ K_p$ | Smanjuje    | Povećava | Povećava     | Smanjuje |
| $\ \ \ \ \ K_p$ | Male promene| Smanjuje | Smanjuje     | Nema promena |

Kao i svaku drugu transfer funkciju, PID kontroler možemo diskretizovati koristeći razne metode. Metoda koja je korišćena u ovom radu je ZOH metoda, *Zero-order hold*, implementirana u biblioteci *Python Control Systems Library*[^3].

[^3]: Python Control Systems Library 0.9.2

U radu su simulirana tri PID kontrolera. Prvi PID kontroler odžava konstantan ugao inklinacije klatna u odnosu na horizontalnu podlogu po kojoj se valjak kreće. Drugi PID kontroler održava konstantnu ugaonu brzinu valjka. Poslednji PID kontroler održava konstantan ugaoni pređeni put valjka. Sva tri PID kontrolera mogu nezavisno da se implementiraju ali ukoliko izlaz jednog PID kontrolera prosleđujemo sledećem PID kontroleru možemo da postignemo isti efekat samo će sistem u ovom slučaju biti manje oscilatoran što doprinosti brzini stabilizacije, *rising time*, i grešci stabilizacije, *steady state error*. Blok dijagram kontrolnog sistema je predstavljen na slici:

![Poprečni presek robota valjka](/images/2022/robot-valjak/prva.drawio2.svg)

> U nastavku se mogu videti grafici dobijeni iz softverskih simulacija:

Reviewer: Prebacite ove rezultate u sekciju o rezultatima

##### Li-ion baterija i LM2598 Stepdown regulator

Napajanje koje se u projektu koristi su 4 redno vezane Samsung 18650-35E Litijum-jonske baterije zbog svog kapaciteta od 5000mAh i maksimalne struje od 2A. Baterije se pune eksternim punjacem pomoću balansirajućeg konektora na njima. Napon sa baterije se sprovodi do LM2598 regulatora gde se napon spušta na 12V kako bi dalje mogao da se dovede do H-mosta, budući da je u trenutnoj konfiguraciji njegov napon ograničen na maksimalnih 12V. Potom se struja iz H-mosta pomocu njegovog integrisanog regulatora spusta na 5V i dovodi do mikrokontrolera i ostalih senzora. Napon baterije se meri na mikrokontroleru kroz naponski razdelnik kako bi se napon od 16.8V spustio na 5V koji su bezbedni za mikrokontroler. Baterija može da radi oko 3.5h u aktivnom režimu robota.

##### DC motor i L298N H-most

Budući da pinovi mikrokontrlera nemaju snagu da pokrenu motor, kao pokretač motora se postavlja H-most koji pomoću signala koji dobija od strane mikrokontrolera pokreće motor. Dva signala koje H-most dobija su digitalni signali: 

1. Koji svojim naponom od 0V ili 5V odredjuje smer i 

> 2. Koji koristeci PWM (Pulse Width Modulation) brzim paljenjem i gašenjem kontroliše napon koji je doveden do motora u opsegu od 0-12V

Reviewer: Može malo bolje objašnjenje. Meni nije jasno šta su ovi signali posle čitanja.

> JGA25370 je DC motor koji radi na naponu od 12V i maksimalnoj struji od 2A. Motor u sebi poseduje reduktor koji umanjuje broj obrtaja motora u sekundi ali povecava obrtni momenat gde je broj obrtaja $RPM_{max} = 250$ obrtaja u minutu i obrtni momenat $T = 1.4kg/cm$. Ovaj motor je jedina komponenta koja pokreće robota.

Reviewer: Ovo važi i za druge brojeve koje koristite.

Ako je nešto iz datašita, okačite link do istog u referencama

## Istraživanje i rezultati

Tokom rada na projektu, uspešno je realizovana hardverska implementacija robota i prikupljeni su podaci o tome kako sistem reaguje na uzastopne odskočne odzive. Tokom istraživanja su izmerena tri odskočna odziva koja redom odgovaraju transfer funkcijama ugla inklinacije klatna u odnosu na podlogu u zavisnosti od napona motora, ugaone brzine klatna u odnosu na podlogu u zavisnosti od ulaznog napona motora i ugaonog pređenog puta robota u zavisnosti od ulaznog napona motora.

> Prilikom dizajniranja klatna valjka centar mase korpe i komponenti (baterije, arduino i ostali moduli) se ne nalazi u osnosimetričnoj ravni korpe već je blago izbačen sa strane što dovodi do toga da nula u merenju inklinacije ugla nije prava nula. Takođe, ovome je potpomogla i činjenica da se na DC motoru nalazi reduktor zbog kog ne postoji prava nula već postoji interval uglova koji mogu biti prava nula. Za male uglove moment zemljine teže na klatno se izjednačava sa momentom sile koju reduktor proizvodi na klatno. Drugim rečima, javalja se mali interval oko prave nule za koje se klatno nalazi u indiferentnoj ravnoteži i u kojoj krajnji položaj zavisi isključivo od početnog impulsa. Ukoliko je veći početni impuls veća je i greška merenja ugla inklinacije. Ovaj fenomen se može predstaviti na grafiku ispod gde se vidi da robot konstantno *driftuje* u jednu stranu. Za dalji rad na ovom robotu treba da se poravna centar mase sistema kao i da se poveća masa klatna kako bi se smanjio opseg uglova koji mogu biti prava nula i na taj način postići preciznija merenja.

Reviewer: 1. "nula u merenju inklinacije ugla nije prava nula" -> "merenje ugla ne bude centrirano oko nule" ("prava nula" zvuči čudno)
2. "Takođe, ovome je potpomogla i činjenica da se na DC motoru nalazi reduktor zbog kog ne postoji prava nula već postoji interval uglova koji mogu biti prava nula" Ovo ima još manje smisla. Takođe, reduktor ne utiče na merenje, on utiče na fiziku koju niste modelovali, a na merenje utiče samo diskretizacija ugla u enkoderu.
3. "Za male uglove moment zemljine teže na klatno se izjednačava sa momentom sile koju reduktor proizvodi na klatno" zašto samo za male uglove? Ovo može da bude pod bilo kojim uglom. Takođe gde je nestao moment usled statičkog trenja?
4. "u kojoj krajnji položaj zavisi isključivo od početnog impulsa" Ovo mi nije najjasnije
5. "Ukoliko je veći početni impuls veća je i greška merenja ugla inklinacije" Kako dinamika robota utiče na merenje uglova? Ovo ne bi trebalo da je tačno
6. "Ovaj fenomen se može predstaviti na grafiku ispod gde se vidi da robot konstantno *driftuje* u jednu stranu" Da li je ovo sigurno tačno? Da nije ovo nešto slično random walk-u, tj. da li ako se eksperiment ponovi dovoljan broj puta, na različitim podlogama pod različitim uglovima vidimo ovaj efekat? Takođe, merenje ugla i ovo što zovete "prava nula" ne utiče na merenje koje ste pokazali zato što nigde ne koristite to merenje.
7. Objasnite ovaj grafik malo bolje, šta je pobuda i gde. Kako je pređeni put robota u radianima?
8. "Za dalji rad na ovom robotu treba da se poravna centar mase sistema kao i da se poveća masa klatna kako bi se smanjio opseg uglova koji mogu biti prava nula i na taj način postići preciznija merenja." - Ne verujem da je ovo zaključak iz onoga što ste napisali i ovog komentara. Ako reduktor unosi nelinearnu komponentu u sistem, to ne može da se izbegne, što znači da treba ili da se zanemari ili doda u algoritam za kontrolu.

Author: 2) Utiče na merenje zato što kada robot miruje klatno nije vertikalno i onda iz očitavanja može da se zaključi da se robot kreće
3) stvar je o tome da iako nema napona na motoru klatno nece mirovati vertikalno već pod nekim uglom. Taj neki ugao ne mora matematički da bude mali, vec da samo postoji ali da nije bas 90stepeni
4) ako je klatno u početku vertikalno i ako ga slabo gurneš on će se pomeriti za neki ugao. Ako ga gurneš malo jače on će se zaustaviti na nekom višem uglu sve dok ga ne gureneš dovoljno jako da ne izađe i te zone.

Author: 5. Ovaj deo nema veze sa robotom dok se kreće već kažem da klatno kada bismo gurnuli ono se ne bi vratilo u prvobitni položaj već bi ostalo pod nekim uglom. Ja sam reko da je taj ugao veći što je impuls veći sve dok impuls ne postane dovoljno velik da zapravo obrne klatno
6. Taj random walk je uvek bio na istoj strani robota i svaki put se pojavljivao. Što se tiče prave nule ona ne utiče ninašta ali nam je predstavljala problem u implementaciji PID-ova zato i stoji tu to je kao naš zaključak šta bi tebalo u buduće da se radi
8. Reduktor unosi nelinearnu komponentu ali je ideja da se klatno oteža i izbalansira kako bi se smanjio efekat reduktora

![Predjeni put robota valjka](/images/2022/robot-valjak/predjeni_put_robota.png)

Druga dva merenja predstavljaju kako u realnom sistemu napon na motoru utiče na ugaonu brzinu klatna i ugao inklinacije klatna. Na drugom grafiku se može jasno videti gore opisan problem. Referentna nula je pomerena za neku vrednost $d\theta$ i razlikuje se od stvarne nule.

![Ugao inklinacije robota valjka](/images/2022/robot-valjak/ugao_inklinacije_klatna.png)

![Ugaona brzina klatna robota valjka](/images/2022/robot-valjak/ugaona_brzina_klatna.png)

> 

Reviewer: Zar niste imali eksperiment sa kontrolom ugla inklinacije i brzine?

title: Robot valjak
summary: Robot u obliku valjka je projekat rađen na letnjem kampu za stare polaznike 2022. godine od Nikole Ristanovića i Filipa Bajraktarija.
---

> ![Grafički apstrakt](/images/2022/robot-valjak/graficki_apstrakt.svg)

Reviewer 2: Izbacite PNG slike koje su zamenjene SVG verzijama iz repozitorijuma, kao što je, između ostalog, `Graficki apstrakt.png`.

[^1]: Control System for a Spherical Robot, Masaki Nagai

##### Horizontalno kretanje

Pre nego što se krene na pravljenje bilo kog hardverskog projekta neophodno je temeljno razumeti mehaniku sistema kako bi mogao da se simulira realni sistem na računaru i na taj način olakša projektovanje i programiranje hardvera. Grana fizike koja se bavi proučavanjem mehanike složenih fizičkih sistema se naziva Lagranžova mehanika. Ona nam služi da složene fizičke sisteme predstavimo matematički isključivo pomću energija za dati sistem. Potencijalne, kinetičke i rotacione kinetičke energije za telo i klatno valjka su date u sledećim izrazima:

$$U_1 = 0$$  
$$U_2 = -M_2gd\cos(\theta _ 1 + \theta _ 2)$$  
$$K_1 = \frac{1}{2}M_1(r\omega _ 1)^2$$  
$$K_2 = \frac{1}{2}M_2((r\omega _ 1 - d\cos(\theta _ 1 + \theta _ 2)(\omega _ 1 + \omega _ 2))^2 + (d\sin(\theta _ 1 + \theta _ 2)(\omega _ 1 + \omega _ 2))^2)$$  
$$T_1 = \frac{1}{2}J_1(\omega _ 1)^2$$  
$$T_2 = \frac{1}{2}J_1(\omega _ 1 + \omega _ 2)^2$$  

> ![Poprečni presek robota valjka](/images/2022/robot-valjak/smer_kretanja.svg)

Reviewer 2: Naziv fajla treba da bude smislen, poput `poprecni-presek.svg`. Je l' ovo vaša slika? Ovaj tekst se teško čita bez zumiranja.

##### DC motor i L298N H-most

Budući da pinovi mikrokontrlera nemaju snagu da pokrenu motor, kao pokretač motora se postavlja H-most koji pomoću signala koji dobija od strane mikrokontrolera pokreće motor. Dva signala koje H-most dobija su digitalni signali: 

1. Koji svojim naponom od 0V ili 5V odredjuje smer i 
2. Koji koristeci PWM (Pulse Width Modulation) brzim paljenjem i gašenjem kontroliše napon koji je doveden do motora u opsegu od 0-12V

Koristeći PWM prividni napon na pinovima mikrokontrolera srazmeran je širini pulsa(Pulse Width) sve dok signal nije potpuno odsutan(0V) ili prisutan(5V). JGA25370 je DC motor koji radi na naponu od 12V i maksimalnoj struji od 2A. Motor u sebi poseduje reduktor koji umanjuje broj obrtaja motora u sekundi ali povecava obrtni momenat gde je broj obrtaja $RPM_{max} = 250$ obrtaja u minutu i obrtni momenat $T = 1.4kg/cm$. Ovaj motor je jedina komponenta koja pokreće robota.

##### Opticki inkrementalni enkoder

> E38S6G5 je optički enkoder koji se koristi za merenje pređenog puta robota. E38S6G5 ima rezoluciju od 600 pulseva po obrtaju osovine što nam daje preciznost od  0.6 stepeni ili ~0.01rad. Princip rada optičkog enkodera je da se analizom izlaznih talasa može odrediti da li se enkoder okrenuo u pozitivnom ili negativnom smeru. U konkretnom slučaju na pin 3 mikrokontrolera je povezan prvi od dva izlaza enkodera, gde se prilikom svake rastuće ivice ovog pina mikrokontroler beleži napon na drugom izlazu enkodera. U slučaju da je napon pozitivan mikrokontroler to belezi kao obrt u pozitivnom smeru, a u slučaju da je napon negativam mikrokontroler to belezi kao obrt u negativnom smeru. Kada za svaki ovaj dogadjaj od nekog brojaša saberemo ili oduzmemo 1 mozemo dobiti ukupni ugao za koji se enkoder zarotirao. Ovime možemo da izračunamo ukupni pređeni put robota i na osnovu toga možemo dobiti trenutnu brzinu.

Reviewer 2: Đoletova sugestija od ranije, latinica, nepotrebna reč.
```suggestion
E38S6G5 je optički enkoder koji se koristi za merenje pređenog puta robota. E38S6G5 ima rezoluciju od 600 pulseva po obrtaju osovine što nam daje preciznost od 0.6 stepeni ili oko 0.01 radijana. Princip rada optičkog enkodera je da se analizom izlaznih talasa može odrediti da li se enkoder okrenuo u pozitivnom ili negativnom smeru. U konkretnom slučaju na pin 3 mikrokontrolera je povezan prvi od dva izlaza enkodera, gde prilikom svake rastuće ivice ovog pina mikrokontroler beleži napon na drugom izlazu enkodera. U slučaju da je napon pozitivan mikrokontroler to beleži kao obrt u pozitivnom smeru, a u slučaju da je napon negativan mikrokontroler to belezi kao obrt u negativnom smeru. Kada za svaki ovaj događaj od nekog brojača saberemo ili oduzmemo 1 možemo dobiti ukupni ugao za koji se enkoder zarotirao. Ovime možemo da izračunamo ukupni pređeni put robota i na osnovu toga možemo dobiti trenutnu brzinu.
```

## Istraživanje i rezultati

Tokom rada na projektu, uspešno je realizovana hardverska implementacija robota i prikupljeni su podaci o tome kako sistem reaguje na uzastopne odskočne odzive. Tokom istraživanja su izmerena tri odskočna odziva koja redom odgovaraju transfer funkcijama ugla inklinacije klatna u odnosu na podlogu u zavisnosti od napona motora, ugaone brzine klatna u odnosu na podlogu u zavisnosti od ulaznog napona motora i ugaonog pređenog puta robota u zavisnosti od ulaznog napona motora.

> Prilikom dizajniranja klatna valjka centar mase korpe i komponenti (baterije, arduino i ostali moduli) se ne nalazi u osnosimetričnoj ravni korpe već je blago izbačen sa strane što dovodi do toga da merenje ugla ne bude centrirano oko nule. Takođe, ovome je potpomogla i činjenica da se na DC motoru nalazi reduktor zbog kog ne postoji prava nula već postoji interval uglova koji mogu biti prava nula. Za male uglove moment zemljine teže na klatno se izjednačava sa momentom sile koju reduktor proizvodi na klatno. Drugim rečima, javalja se mali interval oko prave nule za koje se klatno nalazi u indiferentnoj ravnoteži i u kojoj krajnji položaj zavisi isključivo od početnog impulsa. Ukoliko je veći početni impuls veća je i greška merenja ugla inklinacije. Ovaj fenomen se može predstaviti na grafiku ispod gde se vidi da robot konstantno *driftuje* u jednu stranu. Za dalji rad na ovom robotu treba da se poravna centar mase sistema kao i da se poveća masa klatna kako bi se smanjio opseg uglova koji mogu biti prava nula i na taj način postići preciznija merenja.

Reviewer 2: Veliko slovo.
```suggestion
Prilikom dizajniranja klatna valjka centar mase korpe i komponenti (baterije, Arduino i ostali moduli) se ne nalazi u osnosimetričnoj ravni korpe već je blago izbačen sa strane što dovodi do toga da merenje ugla ne bude centrirano oko nule. Takođe, ovome je potpomogla i činjenica da se na DC motoru nalazi reduktor zbog kog ne postoji prava nula već postoji interval uglova koji mogu biti prava nula. Za male uglove moment Zemljine teže na klatno se izjednačava sa momentom sile koju reduktor proizvodi na klatno. Drugim rečima, javalja se mali interval oko prave nule za koje se klatno nalazi u indiferentnoj ravnoteži i u kojoj krajnji položaj zavisi isključivo od početnog impulsa. Ukoliko je veći početni impuls veća je i greška merenja ugla inklinacije. Ovaj fenomen se može predstaviti na grafiku ispod gde se vidi da robot konstantno *driftuje* u jednu stranu. Za dalji rad na ovom robotu treba da se poravna centar mase sistema kao i da se poveća masa klatna kako bi se smanjio opseg uglova koji mogu biti prava nula i na taj način postići preciznija merenja.
```

> ![Predjeni put robota valjka](/images/2022/robot-valjak/predjeni_put_robota.png)

Reviewer 2: Bolji opis toga šta je na slici za slepe ljude ("grafik zavisnosti X od Y", "grafik na kome su prikazani Z i W"...). Takođe, grafik treba da bude u SVG formatu.

Sledeća dva merenja predstavljaju kako u fizičkom modelu napon na motoru utiče na ugaonu brzinu klatna i ugao inklinacije klatna. Na drugom grafiku se može videti da na početku grafika merenje u stabilnom položaju nije nula. Referentna nula je pomerena za neku vrednost $\Delta\theta$ i razlikuje se od stvarne nule.

![Ugao inklinacije robota valjka](/images/2022/robot-valjak/ugao_inklinacije_klatna.png)

> ![Ugaona brzina klatna robota valjka](/images/2022/robot-valjak/ugaona_brzina_klatna.png)

Reviewer 2: Isto kao iznad.

Takođe, na osnovu transfer funkcija se može zaključiti da je sistem oscilatoran zato su implementirani odgovarajući PID kontroleri. U nastavku se mogu videti grafici stepenastog odziva PID kontrolera dobijeni iz softverskih simulacija:

![PID ugao inklinacije klatna](/images/2022/robot-valjak/PID_ugao_inklinacije_klatna.svg)  
![PID ugaona brzina robota](/images/2022/robot-valjak/PID_ugaona_brzina_robota.svg)  

> ![PID ugaoni predjeni put robota](/images/2022/robot-valjak/PID_ugaoni_predjeni_put_robota.svg)

Reviewer 2: Slično kao iznad.

## Zaključak

> Koristeći pravila Lagranžove mehanike izvedene su jednačine kretanja sistema. Te jednačine opisuje kretanje klatna valjka kao i kretanje samog robota. Jednačine su predstavljene u obliku linearnih diferencijalnih jednačina što dalje omogućava njihovo konvertovanje pomoću Laplasove transformacije u odgovarajuće transfer funkcija. Tako dobijene transfer funkcije nam opisuju kako napon na motoru utiče na ugao inklinacije valjka, ugaonu brzinu valjka i ugaoni pomeraj valjka. Na osnovu transfer funkcija se može zaključiti da je sistem oscilatoran zato su implementirani odgovarajući PID kontroleri koji imaju za ulogu da priguše napomenute oscilacije i u smanje potrebno vreme stabilizacije. U radu su PID kontroleri ulančani i na taj način je omogućeno da *viši* kontroleri brže i sa manjim oscilacijama stabilizuju sistem nego što bi bio slučaj da nije došlo do ulančavanja PID kontrolera.

Reviewer 2: Čudan spoj dve rečenice.
```suggestion
Koristeći pravila Lagranžove mehanike izvedene su jednačine kretanja sistema. Te jednačine opisuje kretanje klatna valjka kao i kretanje samog robota. Jednačine su predstavljene u obliku linearnih diferencijalnih jednačina što dalje omogućava njihovo konvertovanje pomoću Laplasove transformacije u odgovarajuće transfer funkcija. Tako dobijene transfer funkcije nam opisuju kako napon na motoru utiče na ugao inklinacije valjka, ugaonu brzinu valjka i ugaoni pomeraj valjka. Na osnovu transfer funkcija se može zaključiti da je sistem oscilatoran. Zato su implementirani odgovarajući PID kontroleri koji imaju za ulogu da priguše napomenute oscilacije i u smanje potrebno vreme stabilizacije. U radu su PID kontroleri ulančani i na taj način je omogućeno da *viši* kontroleri brže i sa manjim oscilacijama stabilizuju sistem nego što bi bio slučaj da nije došlo do ulančavanja PID kontrolera.
```

Kada je u pitanju hardverska implementacija robota, centar mase klatna robota nije bio u osi geometrijskog centra klatna što je pravilo problem prilikom kalibracije i određivanja ugla inklinacije klatna. Na kalibraciju je dodatno uticalo i postojanje reduktora na DC motoru što je sve ukupno činilo da umesto jedinstvene referentne nule imamo opseg uglova za koje klatno može da se nađe u indiferentnoj ravnoteži.

> Takođe, jedan od ciljeva ovog projekta je bila i numerička identifikacija sistema koja nažalost nije urađena zbog nedostatka vremena kao i gore navedenih problema, ali idalje ostaje kao mogućnost za neki budući rad na ovu ili neku sličnu temu. Pored toga, predlažemo da se u nekom budućem radu da se teg klatna preciznije postavi i izradi od težek materijala kako bi se opseg uglova koji mogu budu referentna nula suzi.

Reviewer 2: Slovne i gramatičke greške.
```suggestion
Takođe, jedan od ciljeva ovog projekta je bila i numerička identifikacija sistema koja nažalost nije urađena zbog nedostatka vremena kao i gore navedenih problema, ali i dalje ostaje kao mogućnost za neki budući rad na ovu ili neku sličnu temu. Pored toga, predlažemo da se u nekom budućem radu teg klatna preciznije postavi i izradi od težeg materijala kako bi se opseg uglova koji mogu budu referentna nula suzi.
```

title: Robot valjak
summary: Robot u obliku valjka je projekat rađen na letnjem kampu za stare polaznike 2022. godine od Nikole Ristanovića i Filipa Bajraktarija.
---

### Uvod

Uvod treba da sadrži sledeće stvari:

> Motivacija iza projekta je bila suprotstavljane ideje popularnog robota - balansera. Ideja je bila da pomoću klatna koje uvek teži da se vrati u stabilan položaj 

Reviewer 3: 1) Za ovo cu morati da razmislim kako da formulisem, ali da li bi dobra ideja bila "Ideja projekta je..." ili slicno
2) Da li umesto balansero mogu da napisem "inverted pendulum robot" posto je to "poznat" termin?

> Motivacija iza projekta je bila suprotstavljane ideje popularnog robota - balansera. Ideja je bila da pomoću klatna koje uvek teži da se vrati u stabilan položaj 

Reviewer: 1. Da to je ok
2. Nije dovoljno poznat za uvod

> robot može biti konstruisan sa manjom preciznošću što bi ubrzalo proces pravljenja u slučaju masovne proizvodenje, kao i vreme dolaska do prvog funkcionalnog prototipa, budući da robot može da radi uz bilo kakav (ne)kontrolisan pogon motora. Takođe zbog oblika robota sve vitalne komponente se sadržane unutar konture, čineći robota izdržljivijim i jednostavnijim za popravku.

Reviewer 3: 1) Zato sto za razliku od balansera komponente mogu da budu manje precizne i rabot ce nastaviti da radi, ali uz manje performanse.
2) Zato sto se konstrukcija sastoji iz delova koji su jednostavni za proizvodnju, mogu da imaju manju preciznost, manje kontrolisan sistem i izdrzljiv je zbog svoje konstrukcije, sto sve doprinsi jefinijoj ceni izrade.

> robot može biti konstruisan sa manjom preciznošću što bi ubrzalo proces pravljenja u slučaju masovne proizvodenje, kao i vreme dolaska do prvog funkcionalnog prototipa, budući da robot može da radi uz bilo kakav (ne)kontrolisan pogon motora. Takođe zbog oblika robota sve vitalne komponente se sadržane unutar konture, čineći robota izdržljivijim i jednostavnijim za popravku.

Reviewer: 1. To je dobro, dodajte u tekst.
2. To što se nešto lako pravi ne znači da treba. Pitanje je "čemu ovo služi?". Zamisli da prodaješ ovu ideju nekoj velikoj firmi i budeš u fazonu "Ovo je mnogo dobar robot, ne radi ništa, ali zato može da se napravi 10 puta jeftinije nego pre"
3.

> Reference su skoro ne postojeće budući da se ideja o robotu dobila bez drugog uzora/slične ideje.

Reviewer 3: Reference ce biti dodate cim ih skupimo se u jedan fajl

Za razliku od klasične Njutnove mehanike za Lagranžovu mehaniku je potrebno definisati odgovarajuću Ojler-Lagranžove funkcije, koja se definiše na sledeći način:
$$L = K_1 + K_2 + T_1 + T_2 - U_1 - U_2$$
Primenivši Ojler-Lagranžovu jednačinu na dva sistema, klatno i valjak,
$$\frac{d}{dt}\left(\frac{\partial L}{\partial \omega _ 1}\right)\ -\ \frac{\partial L}{\partial \omega _ 1}\ = \ -T + T_f$$
$$\frac{d}{dt}\left(\frac{\partial L}{\partial \omega _ 2}\right)\ -\ \frac{\partial L}{\partial \omega _ 2}\ = \ T$$
dobijaju se sledeće jednačine kojima je u potpunosti opisana dinamika sistema:
$$-T + T_f\ =\ \alpha_1(J_1 + J_2 + M_1r^2 + M_2r^2 + M_2d^2 - 2M_2rd)\ +\ \alpha_2(J_2 - M_2rd + M_2d^2)\ +\ M_2gd(\theta_1 + \theta_2)$$
$$T\ =\ \alpha_1(J_1 - M_2rd + M_2d^) + \alpha_2(J_2 + M2d^2) + M_2gd(\theta_1 + \theta_2)$$
gde su iskorišćene aproksimacije za male uglove i male ugaone brzine.

#### Simulacije

Simulacije služe da se na što realističniji način predstavi realni svet i dobije bolji uvid u to kako će se sistem ponašati. Pored osećaja za sistem, simulacije omogućavaju lakše testiranje softvera kao i testiranje sistema u različitim prirodnim uslovima.

> ##### Transfer funkcije

Author: Jednačine transfer funkcija su dodate ali nisam siguran da li treba da stavim u matričnom obliku pošto osim da kažem ovo vam je matrica ne znam da objasnim ideju iza toga

Transfer funkcija (funkcija prenosa) je definisana kao Laplasova transformacija izlazne promenjljive i Laplasove transformacije ulazne promenjljive pod pretpostavkom da su svi početni uslovi jednaki nuli.
$$G(s)\ =\ \frac{C(s)}{R(s)}$$
Način na koji se formira transfer funkcija kontrolnog sistema je sledeća:

1. Formiraju se jednačine sistema
2. Uzimaju se Laplasove transformacije jednačina, pretpostavljajući da su početni uslovi jednaki nuli
3. Određuju se ulazi i izlazi sistema
4. Na kraju se uzima odnos Laplasove transformacije izlaza i ulaza što je tražena transfer funkcija

Takođe, važno je naglasiti da izlaz i ulaz nekog sistema ne mora nužno da bude istih fizičkih dimenzija.

U radu su korišćene tri transfer funkcije koje se uz data pravila mogu izvesti iz fizičkih jednačina za horizontalno kretanje valjka. To su transfer funkcija ugla inklinacije klatna valjka i napona na motoru, transfer funkcija ugaone brzine klatna valjka i napona na motoru i, na kraju, ugaona brzina valjka i napona motora.

TODO: Analiza stabilnosti transfer funkcija

##### PID simulacije

> Kako bi se razumeo PID kontroler potrebno je definisati odredjene termine. *Željena vrednost* je obično vrednost koju unosi korisnik. Na primer, za sistem za grejanje to bi bila željena temperatura prostorije. *Izlaz* je kontrolisana vrednost PID kontrolera. U automobilu bi to bila količina goriva koja je potrebna da bi se održavala konstantna brzina. *Greška* je vrednost koju koristi PID kontroler da bi odredio kako da manipuliše izlazom kako bi se dostigla željena vrednost. Greška se računa kao razlika željene vrednosti i trenutne vrednosti sistema.

Reviewer: 1. Promenite uvod ovoga. Zašto bi neko hteo da razume PID kontroler?
1. Možda ove stvari poput "ulaz" i "izlaz" da objasnite na početku dinamike, kada objašnjavate transfer funkcije.
1. Ovo za gorivo i automobil mi nije baš najjasnije
1. Bilo bi super ako ubacite šemu kontrolera

U radu su simulirana tri PID kontrolera. Prvi PID kontroler odžava konstantan ugao inklinacije klatna u odnosu na horizontalnu podlogu po kojoj se valjak kreće. Drugi PID kontroler održava konstantnu ugaonu brzinu valjka. Poslednji PID kontroler održava konstantan ugaoni pređeni put valjka. Blok dijagram kontrolnog sistema je predstavljen na slici:

> 

Reviewer: Fali blok diagram

> Kao što je u prethodnom poglavlju rečeno, transfer funkcija predstavlja odnos između signala upravljačkog sistema i ulaznog signala, za sve moguće ulazne vrednosti. Ukoliko je transfer funkcija predstavljena preko Laplasovih transformacija onda je takva funkcija prenosa definisana na kontinualnom domenu. Takav oblik transfer funkcija je od koristi prilikom teorijskih analiza kao što je analiza polova i nula. Nazalost, u praksi nam treba diskretna reprezentacija sistema što proizilazi iz toga da mikrokontroler rade isključivo nad diskretnim domenom vrednosti. Postoje razni algoritmi koji vrše željenu konverziju. Jedan od njih, koji je i koriščen u radu, *Zero-order Hold* metoda 

Reviewer: 1. Nemojte da ponavljate definicije
2. "Kao što je u prethodnom poglavlju rečeno" izbaciti (isto ako i pre komentar što sam dao)
3. "Ukoliko je transfer funkcija predstavljena preko Laplasovih transformacija onda je takva funkcija prenosa definisana na kontinualnom domenu" - Ovo je malo bezveze napisano. Bolje objasnite ovo u trenutku kada pišete o transfer funkcijama, ako vam je to bitno, pa posle pišite "kontinualna transfer funkcija" i "diskretna transfer funkcija"
4. "Takav oblik transfer funkcija je od koristi prilikom teorijskih analiza kao što je analiza polova i nula." Vi ovo niste radili. Pored toga, niste objasnili šta su polovi i nule.
5. Zašto "Nažalost"? To je dobra stvar da se diskretizuje, zato što možemo da pravimo kompleksnije kontrolere na jednostavniji način.
6. Objasnite šta je "Zero-order hold" i zašto ste ga koristili

Author: Stavke od jedan do pet su odradjene. Što se tiče ZOHa je ista situacija kao i za matričnu transfer funkciju, ja nigde nisam implementirana ZOH funkcija već je samo prosledjena kao parametar Control funkciji.

> ## Istraživanje i rezultati

Reviewer: Fali vam diskusija o komplementarnom filteru
